src/CoreWf/VoidResult.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System.Threading.Tasks;
+
+namespace System.Activities
+{
+    /// <summary>
+    /// Represents a void type, since <see cref="System.Void"/> is not a valid return type in C#.
+    /// </summary>
+    /// <remarks>Based on the Unit type from MediatR.</remarks>
+    internal readonly struct VoidResult : IEquatable<VoidResult>, IComparable<VoidResult>, IComparable
```

Comment:
`VoidResult` is based on the [Unit](https://github.com/jbogard/MediatR/blob/master/src/MediatR/Unit.cs) type from Mediatr, licensed under the [Apache 2.0](https://github.com/jbogard/MediatR/blob/master/LICENSE) license.

The only change made to the type was its renaming to `VoidResult`, which I felt better showed its intent.

Reply:
I understand the idea, but given the fact that we don't really use much of this `void` type, I think we can simply use `object` instead. It's just an implementation detail, we can always introduce such a type if the need arises.